btree/btree.c

```diff
@@ -1388,55 +1912,55 @@ static bool ff_isfit(struct slot *slot)
 {
 	struct ff_head *h = ff_data(slot->s_node);
 
-	M0_PRE(ff_invariant(slot->s_node));
 	M0_PRE(ff_rec_is_valid(slot));
-	return h->ff_ksize + h->ff_vsize >= ff_space(slot->s_node);
+	return h->ff_ksize + h->ff_vsize <= ff_space(slot->s_node);
 }
 
 static void ff_done(struct slot *slot, struct m0_be_tx *tx, bool modified)
 {
-	M0_PRE(ff_invariant(slot->s_node));
 }
 
 static void ff_make(struct slot *slot, struct m0_be_tx *tx)
 {
-	struct nd      *node  = slot->s_node;
-	struct ff_head *h     = ff_data(node);
-	int             rsize = h->ff_ksize + h->ff_vsize;
-	void           *start = ff_key(node, slot->s_idx);
+	const struct nd *node  = slot->s_node;
+	struct ff_head  *h     = ff_data(node);
+	int              rsize = h->ff_ksize + h->ff_vsize;
+	void            *start = ff_key(node, slot->s_idx);
 
-	M0_PRE(ff_invariant(node));
 	M0_PRE(ff_rec_is_valid(slot));
 	M0_PRE(ff_isfit(slot));
 	memmove(start + rsize, start, rsize * (h->ff_used - slot->s_idx));
 	h->ff_used++;
 }
 
-static void ff_find(struct slot *slot, const struct m0_btree_key *key)
+static bool ff_find(struct slot *slot, const struct m0_btree_key *find_key)
 {
 	struct ff_head *h = ff_data(slot->s_node);
-	int i = 0;
-	int j = h->ff_used;
-	M0_PRE(ff_invariant(slot->s_node));
-	M0_PRE(key->k_data.ov_vec.v_count[0] == h->ff_ksize);
-	M0_PRE(key->k_data.ov_vec.v_nr == 1);
+	int             i = -1;
+	int             j = h->ff_used;
```

Comment:
int j = node_count(slot->s_node)
because, at internal node, we don't want to consider last key. last key can be null or any random value